plugins/providers/shield/provider.go

```diff
@@ -0,0 +1,267 @@
+package shield
+
+import (
+	"github.com/mitchellh/mapstructure"
+	pv "github.com/odpf/guardian/core/provider"
+	"github.com/odpf/guardian/domain"
+	"github.com/odpf/salt/log"
+)
+
+type provider struct {
+	pv.UnimplementedClient
+	pv.PermissionManager
+
+	typeName string
+	Clients  map[string]ShieldClient
+	logger   log.Logger
+}
+
+func (p *provider) GetAccountTypes() []string {
+	return []string{
+		AccountTypeUser,
+	}
+}
+
+func NewProvider(typeName string, logger log.Logger) *provider {
```

Comment:
Can you add documentation for this shield provider with example policy and provider configuration